### PR TITLE
Moved from InferenceApi to InferenceClient to avoid deprecation warning for huggingface_hub

### PR DESCRIPTION
**Pull request by keenborder786**
Updated huggingface_hub llm to use the new InferenceClient to avoid deprecation warning. 

Resolves the issue: #10483

@baskaryan, @eyurtsev, @hwchase17, @rlancemartin.

### CODE REVIEW

File: libs/langchain/langchain/llms/huggingface_hub.py
Comment (baskaryan) [inline libs/langchain/langchain/llms/huggingface_hub.py:113]: getting an error here, InferenceClient.post seems to return bytes

Comment (keenborder786) [reply libs/langchain/langchain/llms/huggingface_hub.py:113]: @baskaryan will fix it.

Comment (keenborder786) [reply libs/langchain/langchain/llms/huggingface_hub.py:113]: @baskaryan  I double-checked the response is a JSON. I am not so sure why you are saying it is bytes.

Comment (keenborder786) [reply libs/langchain/langchain/llms/huggingface_hub.py:113]: @baskaryan Can you please post your error?

Comment (keenborder786) [reply libs/langchain/langchain/llms/huggingface_hub.py:113]: ?

Comment (keenborder786) [reply libs/langchain/langchain/llms/huggingface_hub.py:113]: okay, I will review it and fix it.

Comment (keenborder786) [reply libs/langchain/langchain/llms/huggingface_hub.py:113]: @easelify @baskaryan I again the ran the following code:
```python

from langchain.llms import HuggingFaceHub
from langchain.prompts import PromptTemplate
from langchain.chains import LLMChain
import os
os.environ['HUGGINGFACEHUB_API_TOKEN'] = ''
prompt = PromptTemplate(template = 'Please answer the following question: {question}',input_variables=['question'])

llm_chain = LLMChain(prompt=prompt,
                     llm=HuggingFaceHub(
                         repo_id="google/flan-t5-small",
                         model_kwargs={
                             "temperature": 0,
                             "max_length": 64
                         }
                     )
                     )

question = "What is the capital of England?"

print(llm_chain.run(question))

```

and got no errors. The `InferenceClient` post method returns a Python Request object which is converted to json object therefore I don't get it why you guys are getting a bytes object. It might be due to different version for huggingface_hub.

Can I ask which version of `huggingface_hub` are you guys using? My version is as follows
```
ame: huggingface-hub
Version: 0.15.1
Summary: Client library to download and publish models, datasets and other repos on the huggingface.co hub
Home-page: https://github.com/huggingface/huggingface_hub
Author: Hugging Face, Inc.
Author-email: julien@huggingface.co
License: Apache

```

Comment (keenborder786) [reply libs/langchain/langchain/llms/huggingface_hub.py:113]: @easelify ???

**Comment by Leah-Mac** [comment]
Sorry if this is a dumb question, I am new to this. Is this update/fix on the newest version of langchain (0.0.303)? After updating I still get the inference api depreciation warning when running a huggingface_hub model:
\venv\Lib\site-packages\huggingface_hub\utils\_deprecation.py:127: FutureWarning: '__init__' (from 'huggingface_hub.inference_api') is deprecated and will be removed from version '0.19.0'. `InferenceApi` client is deprecated in favor of the more feature-complete `InferenceClient`. Check out this guide to learn how to convert your script to use it: https://huggingface.co/docs/huggingface_hub/guides/inference#legacy-inferenceapi-client.
  warnings.warn(warning_message, FutureWarning)

**Comment by keenborder786** [comment]
@Leah-Mac This PR is still Work in Progress, once this will get merge then the warning will no longer be visible to you.

File: libs/langchain/langchain/llms/huggingface_hub.py
Comment (keenborder786) [reply libs/langchain/langchain/llms/huggingface_hub.py:113]: @easelify waiting for your reply.

Comment (easelify) [reply libs/langchain/langchain/llms/huggingface_hub.py:113]: Sorry, I have been offline for a while recently. The version I am using is 0.17.2

Comment (mspronesti) [reply libs/langchain/langchain/llms/huggingface_hub.py:113]: Don't you need to decode the output of `post` first? Also, does `bytes` have a `json()` method?
I suppose the correct thing to do here is:

```py
response = self.client.post(
   json={"inputs": prompt, "params": params}, task=self.task
)

data = json.loads(response.decode())
```

**Review by mspronesti — CHANGES_REQUESTED** [review]

**Comment by mspronesti** [comment]
Any updates on this?

**Pull request closed without merging**
